src/main/java/com/google/firebase/auth/internal/KeyManagers.java

```diff
@@ -0,0 +1,108 @@
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.client.util.Clock;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.internal.NonNull;
+
+/**
+ * A utility for initializing and keeping tack of the various public key manager instances
+ * used by {@link com.google.firebase.auth.FirebaseAuth}.
+ */
+public class KeyManagers {
+
+  private final GooglePublicKeysManager idTokenKeysManager;
+  private final GooglePublicKeysManager sessionCookieKeysManager;
+
+  private KeyManagers(Builder builder) {
+    this.idTokenKeysManager = checkNotNull(builder.idTokenKeysManager);
+    this.sessionCookieKeysManager = checkNotNull(builder.sessionCookieKeysManager);
+  }
+
+  /**
+   * Returns the key manager that should be used for ID token verification.
+   */
+  GooglePublicKeysManager getIdTokenKeysManager() {
+    return idTokenKeysManager;
+  }
+
+  /**
+   * Returns the key manager that should be used for session cookie verification.
+   */
+  GooglePublicKeysManager getSessionCookieKeysManager() {
+    return sessionCookieKeysManager;
+  }
+
+  /**
+   * Initialize a new set of key managers for the specified app using the given clock.
+   *
+   * @param app A {@link FirebaseApp} instance.
+   * @param clock A {@code Clock} to be used with Google API client.
+   * @return A new {@link KeyManagers} instance.
+   */
+  public static KeyManagers getDefault(@NonNull FirebaseApp app, @NonNull Clock clock) {
+    HttpTransport transport = app.getOptions().getHttpTransport();
+    return getDefault(transport, clock);
+  }
+
+  @VisibleForTesting
+  static KeyManagers getDefault(HttpTransport transport, Clock clock) {
+    return new Builder()
+        .setIdTokenKeysManager(createPublicKeysManager(
+            transport, clock, FirebaseTokenVerifier.ID_TOKEN_CERT_URL))
+        .setSessionCookieKeysManager(createPublicKeysManager(
+            transport, clock, FirebaseTokenVerifier.SESSION_COOKIE_CERT_URL))
+        .build();
+  }
+
+  private static GooglePublicKeysManager createPublicKeysManager(
+      HttpTransport transport, Clock clock, String certUrl) {
+    return new GooglePublicKeysManager.Builder(transport, new GsonFactory())
+        .setClock(clock)
+        .setPublicCertsEncodedUrl(certUrl)
+        .build();
+  }
+
+  public static class Builder {
+
+    private GooglePublicKeysManager idTokenKeysManager;
+    private GooglePublicKeysManager sessionCookieKeysManager;
+
+    public Builder setIdTokenKeysManager(
+        GooglePublicKeysManager idTokenKeysManager) {
+      this.idTokenKeysManager = idTokenKeysManager;
+      return this;
+    }
+
+    public Builder setSessionCookieKeysManager(
+        GooglePublicKeysManager sessionCookieKeysManager) {
+      this.sessionCookieKeysManager = sessionCookieKeysManager;
+      return this;
+    }
+
+    public KeyManagers build() {
```

Comment:
Since this is internal, would it be possible to not use the Builder pattern here? This is otherwise going to be used as an example for why Java is bloated :)